### ISSUE TITLE
False negative in IgnoredReturnValue

### ISSUE THREAD

**Issue by BraisGabin** [detekt/detekt#3366]
## Expected Behavior
`IgnoredReturnValue`shoudl raise two issues in this code:

```kotlin
package test

annotation class CheckReturnValue

@CheckReturnValue
fun listA() = listOf("hello")

fun foo(bar: String?) {
    val requestOptions = "".apply {
        bar?.let {
            listA() // (1)
        }
        bar?.let {
            listA() // (2)
        }
    }
}
```

## Observed Behavior
Should only raise one (the second one `(2)`)

## Context
I know that this code is really strange but it's a simplification of what I have in production that have the same problem

## Your Environment
* Version of detekt used: 1.15.0

**Comment by schalkms** [comment]
Code (1) and code (2) are equal. Is this expected or a copy and paste mistake? Does the checked return value depend on the order of the program code?

**Comment by cortinico** [comment]
> Should only raise one (the second one `(2)`)

Also why it should only raise the second? To me it looks like you're not using the two values.

**Comment by BraisGabin** [comment]
You are right. This case have no sense. It should raise both... I don't remember the complex case here so I'm closing this. I'll reopen it if I remember the reason of this issue.
